### PR TITLE
Basic Docs setup

### LINKED ISSUE #5: Add readthedocs.io documentation
We should have a publicly available documentation for the package and all its tools.

### PR DESCRIPTION
**Pull request by maxfischer2781**
This pull request adds basic documentation. Changes include:

* [x] basic sphinx docs,
* [x] automatic readthedocs builds.

There is a test build at [readthedocs](https://change-log.readthedocs.io/en/feature-docs/index.html). Closes #5.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'change-log'
+copyright = '2019, Eileen Kuehn, Max Fischer'
+author = 'Eileen Kuehn, Max Fischer'
```
Comment (eileen-kuehn) [inline docs/conf.py]: If you exchange those two lines you can even include the ``author`` into the copyright ;)

File: docs/subtool/log.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+versions tie together all changes that are released together.
+
+Changelogs are based on `keep a changelog`_ and `semantic versioning`_.
+
+Fragments
+---------
+
+Fragments are YAML files that contain meta-data
+and human-readable descriptions of individual changes.
+Each fragment file contains a mappings that must contain
```
Comment (eileen-kuehn) [inline docs/subtool/log.rst]:
```suggestion
Each fragment file contains a mapping that must contain
```

File: docs/subtool/log.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+the fields ``category``, ``summary``, and ``description`` and optionally the fields
+``pull requests`` and ``issues``; a unique naming convention of files such as
+``<first PR>.<topic>.yaml`` is recommended.
+Both ``summary`` and ``description`` fields are interpreted as reStructured Text.
+
+.. code:: YAML
+
+    # file `39.line_format.fixes.yaml`
+    # any of 'added', 'changed', 'fixed', 'deprecated', 'removed', 'security'
```
Comment (eileen-kuehn) [inline docs/subtool/log.rst:29]: Please consider to put these options into the description text not just the example code.

File: docs/subtool/log.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      - 39
+      - 44
+    # issues solved by this change
+    issues:
+      - 42
+    # long description of changes
+    description: |
+      The Line Protocol implementation has been extended to remove cases that
+      previously led to illegal output. ``None`` values are
+      forbidden, and strings are escaped in field values, tags, and measurements.
```
Comment (eileen-kuehn) [inline docs/subtool/log.rst]: What do you think about using an example that does match this repository? For example you could include this PR itself. So if anybody tries to interpret this also finds the referenced PR and so on.

File: docs/subtool/log.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+The ``version`` is optional and in most cases does not need manual definitions.
```
Comment (eileen-kuehn) [inline docs/subtool/log.rst]:
```suggestion
The ``version`` is optional and in most cases does not need manual definition.
```

File: docs/subtool/log.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+Unversioned fragments belong to the "next" release, and
```
Comment (eileen-kuehn) [inline docs/subtool/log.rst]:
```suggestion
Unversioned fragments belong to the *next* release, and
```

**Review by eileen-kuehn — CHANGES_REQUESTED** [review]
Great thing, we should consider having something like this in a project template :)

However, I left some minor comments you should have a look at before merging.

File: docs/subtool/log.rst (hunk 4 above)
Comment (maxfischer2781) [reply docs/subtool/log.rst]: I think that's a great idea!

File: docs/subtool/log.rst (hunk 3 above)
Comment (maxfischer2781) [reply docs/subtool/log.rst:29]: Documented the expected choices for ``category``.

File: docs/subtool/log.rst (hunk 4 above)
Comment (maxfischer2781) [reply docs/subtool/log.rst]: Now using this PR and a new issue #5 as example.

File: docs/conf.py (hunk 1 above)
Comment (maxfischer2781) [reply docs/conf.py]: Dome.

**Review by eileen-kuehn — APPROVED** [review]
Great, especially the self-reference 👍

**Pull request merged**